src/app/shared/services/authors.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+import { AppConfig } from 'src/app/app.config';
+
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { LoadingService } from './loading.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class AuthorsService {
+    private modelEndpoint = AppConfig.apiUrl + '/authors';
+
+  constructor(
+      private httpClient: HttpClient,
+      private loadingService: LoadingService
+  ) { }
+
+    findAuthors(serachWord): Observable<any> {
```

Comment:
The method name should be smth like "search" as this is part of the authorsService is pointless to have method findAuthors - this.authorsService.findAuthors, it's better this.authorsService.search